File: 2021m./uzd1./sprendimai/uzd1.py
def getTimeRan(mHourStart, mMinStart, mHourEnd, mMinEnd, 
               eHourStart, eMinStart, eHourEnd, eMinEnd):
    minutesRan = (mHourEnd * 60 + mMinEnd - mHourStart * 60 - mMinStart +
                  eHourEnd * 60 + eMinEnd - eHourStart * 60 - eMinStart )
    return minutesRan

def main():
    data = open("U1.txt", "r")
    output = open("U1rez.txt", "w")

    minTime = 10000000
    minDays = []

    for i in range(int(next(data))):
        dayData = next(data).split(" ")
        dayData = [int(o) for o in dayData]
        if(dayData[1:5] == [0,0,0,0] or dayData[5:9] == [0,0,0,0]):
            continue
        minutesRan = getTimeRan(dayData[1], dayData[2], dayData[3], dayData[4],
                                dayData[5], dayData[6], dayData[7], dayData[8])
        if(minutesRan == minTime):
            minDays.append(str(dayData[0]))
        elif(minutesRan < minTime):
            minTime = minutesRan
            minDays = [str(dayData[0])]

    

    output.write("Minimalus laikas\n")
    output.write(f"{minTime}\n")
    output.write("Dienos\n")
    output.write(" ".join(minDays))

    data.close()
    output.close()

if __name__ == "__main__":
    main()

File: 2021m./uzd1./sprendimai/uzd1.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>

using namespace std;

int getTimeRan(int mHourStart, int mMinStart, int mHourEnd, int mMinEnd, 
               int eHourStart, int eMinStart, int eHourEnd, int eMinEnd) {
    int minutesRan = (mHourEnd * 60 + mMinEnd - mHourStart * 60 - mMinStart +
                      eHourEnd * 60 + eMinEnd - eHourStart * 60 - eMinStart);
    return minutesRan;
}

int main() {
    ifstream data("U1.txt");
    ofstream output("U1rez.txt");

    int minTime = 10000000;
    vector<int> minDays;

    int numDays;
    data >> numDays;

    for (int i = 0; i < numDays; ++i) {
        int dayData[9];
        for (int j = 0; j < 9; ++j) {
            data >> dayData[j];
        }
        if (dayData[1] == 0 && dayData[2] == 0 && dayData[3] == 0 && dayData[4] == 0)
            continue;
        if (dayData[5] == 0 && dayData[6] == 0 && dayData[7] == 0 && dayData[8] == 0)
            continue;
        int minutesRan = getTimeRan(dayData[1], dayData[2], dayData[3], dayData[4],
                                    dayData[5], dayData[6], dayData[7], dayData[8]);
        if (minutesRan == minTime) {
            minDays.push_back(dayData[0]);
        } else if (minutesRan < minTime) {
            minTime = minutesRan;
            minDays = {dayData[0]};
        }
    }

    output << "Minimalus laikas\n";
    output << minTime << "\n";
    output << "Dienos\n";
    for (const int& day : minDays) {
        output << day << " ";
    }

    data.close();
    output.close();

    return 0;
}

